File: backend/app/schemas/expedition.py
from typing import List, Optional
from pydantic import BaseModel, ConfigDict, Field


class ExpeditionMilestone(BaseModel):
    id: str
    title: str
    location: str
    date: str
    status: str
    description: str


class ExpeditionResponse(BaseModel):
    id: str
    code: Optional[str] = None
    name: str
    short_name: str = Field(..., validation_alias="shortName", serialization_alias="shortName")
    season: str
    status: str
    start_date: str = Field(..., validation_alias="startDate", serialization_alias="startDate")
    end_date: str = Field(..., validation_alias="endDate", serialization_alias="endDate")
    leader: str
    vessel: str
    personnel_count: int = Field(..., validation_alias="personnelCount", serialization_alias="personnelCount")
    cargo_tonnage: float = Field(..., validation_alias="cargoTonnage", serialization_alias="cargoTonnage")
    active_missions_count: int = Field(..., validation_alias="activeMissionsCount", serialization_alias="activeMissionsCount")
    overall_readiness_pct: float = Field(..., validation_alias="overallReadinessPct", serialization_alias="overallReadinessPct")
    milestones: List[ExpeditionMilestone]

    model_config = ConfigDict(
        populate_by_name=True,
        from_attributes=True,
    )

File: frontend/src/services/expedition.ts
import { apiClient } from "./apiClient";
import { Expedition, Station } from "@/types";
import { stationsService } from "./stations";


export const expeditionService = {
  async getActiveExpedition(): Promise<Expedition> {
    const exp = await apiClient.get<any>("/expeditions/active");
    return {
      id: exp.id || exp.code || "ISEA-46",
      name: exp.name || "46th Indian Scientific Expedition to Antarctica",
      shortName: exp.shortName || exp.short_name || "46th ISEA",
      season: exp.season || "2026-2027 Austral Season",
      status: exp.status || "ACTIVE",
      startDate: exp.startDate || exp.start_date || "2026-11-15",
      endDate: exp.endDate || exp.end_date || "2027-04-10",
      leader: exp.leader || "Dr. Arvind Sharan (Scientist 'G', NCPOR)",
      vessel: exp.vessel || "MV Vasiliy Golovnin (Charter Icebreaker)",
      personnelCount: exp.personnelCount ?? exp.personnel_count ?? 0,
      cargoTonnage: exp.cargoTonnage ?? exp.cargo_tonnage ?? 0,
      activeMissionsCount:
        exp.activeMissionsCount ?? exp.active_missions_count ?? 0,
      overallReadinessPct:
        exp.overallReadinessPct ?? exp.overall_readiness_pct ?? 0,
      milestones: exp.milestones || [],
    };
  },

  async getExpeditionById(id: string): Promise<Expedition | null> {
    return await apiClient.get<Expedition>(`/expeditions/${id}`);
  },

  async getStations(): Promise<Station[]> {
    return await stationsService.getAllStations();
  },

  async getStationById(id: string): Promise<Station | undefined> {
    return await stationsService.getStationById(id);
  },
};
